src/main/java/com/cloudbees/hudson/plugins/folder/config/AbstractFolderConfiguration.java

```diff
@@ -30,27 +34,47 @@ public static AbstractFolderConfiguration get() {
 
     @DataBoundConstructor
     public AbstractFolderConfiguration() {
-        load();
-        if (healthMetrics == null) {
+        this.load();
+    }
+
+    /**
+     * Auto-configure the default metrics after all plugins have been loaded.
+     */
+    @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED, before = InitMilestone.JOB_LOADED)
+    public static void autoConfigure() {
+        AbstractFolderConfiguration abstractFolderConfiguration = AbstractFolderConfiguration.get();
+        if (abstractFolderConfiguration.healthMetrics == null) {
             List<FolderHealthMetric> metrics = new ArrayList<>();
             for (FolderHealthMetricDescriptor d : FolderHealthMetricDescriptor.all()) {
                 FolderHealthMetric metric = d.createDefault();
                 if (metric != null) {
                     metrics.add(metric);
                 }
             }
-            healthMetrics = new DescribableList<FolderHealthMetric, FolderHealthMetricDescriptor>(this, metrics);
+            abstractFolderConfiguration.setHealthMetrics(new DescribableList<FolderHealthMetric,
+                    FolderHealthMetricDescriptor>(abstractFolderConfiguration, metrics));
         }
     }
-    
+
     @Nonnull
     public List<FolderHealthMetric> getHealthMetrics() {
         return healthMetrics == null ? Collections.emptyList() : healthMetrics;
     }
-    
+
     @DataBoundSetter
     public void setHealthMetrics(List<FolderHealthMetric> healthMetrics) {
         this.healthMetrics = healthMetrics;
         save();
     }
-}
\ No newline at end of file
+
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject json) {
+        if(json.containsKey("healthMetrics")) {
+            req.bindJSON(this, json);
+            this.save();
+        } else {
+            this.setHealthMetrics(Collections.emptyList());
+        }
```

Comment:
Missed `save()` ?

Maybe simpler to 

```java
this.setHealthMetrics(Collections.emptyList());
req.bindJSON(this, json);
this.save();
```

?

Reply:
ah I see now the call to `save()` in `setHealthMetrics`

Reply:
yeah this would `save()` twice